### PR TITLE
Apply canonical service yaml (unless explicitly disabled)

### PR DESCRIPTION
**Pull request by djazayeri**

### CODE REVIEW

**Comment by djazayeri** [comment]
Do not merge until we get improved wording from the team that owns the canonical service component.

File: scripts/asm-installer/install_asm
Comment (jasonwzm) [inline scripts/asm-installer/install_asm:882]: Add verification on the deployment? kubectl wait?

Comment (djazayeri) [reply scripts/asm-installer/install_asm:882]: Good point, PTAL.

Comment (jasonwzm) [reply scripts/asm-installer/install_asm:882]: Also can we add a warning or error when the wait fails. Similar to what istioctl does during install.

Comment (djazayeri) [reply scripts/asm-installer/install_asm:882]: I agree that's nice-to-have, but we don't do that elsewhere in this script.

File: scripts/asm-installer/install_asm
Comment (zerobfd) [inline scripts/asm-installer/install_asm:880]: `if [[ "${DISABLE_CANONICAL_SERVICE}" -ne 1 ]]`

Very intuitively, != and = are string comparison operators in shell, you use -eq and -ne for numeric comparisons.

https://tldp.org/HOWTO/Bash-Prog-Intro-HOWTO-11.html

File: scripts/asm-installer/install_asm
Comment (zerobfd) [inline scripts/asm-installer/install_asm:881]: It may be worth calling out here that it's going to the asm_system namespace. Someone might delete it if they don't recognize it otherwise.

**Review by zerobfd — COMMENTED** [review]
lgtm after the comparison fix, will approve after the wording update as per your earlier comment

File: scripts/asm-installer/install_asm
Comment (djazayeri) [reply scripts/asm-installer/install_asm:899]: Will add.

File: scripts/asm-installer/install_asm
Comment (djazayeri) [reply scripts/asm-installer/install_asm:898]: Will change.

**Review by zerobfd — APPROVED** [review]

**Pull request merged**
